### PR TITLE
fix(json): use char boundary when truncating long string values

### PR DESCRIPTION
**Pull request by swithek**
## Summary

- Fix UTF-8 panic in `rtk json` when a multi-byte char crosses byte 77 of a string value (e.g. JSON containing Japanese, Cyrillic, accented Latin, or emoji).
- `compact_json` now rounds the truncation index down to the nearest valid char boundary with `str::floor_char_boundary`, matching the existing pattern in `pipe_cmd::auto_detect_filter`.
- Add regression test covering the multi-byte boundary case.

## Test plan

- [x] `cargo fmt --all && cargo clippy --all-targets && cargo test`
- [x] Manual testing: `rtk <command>` output inspected

  ```sh
  python3 -c 'import json; print(json.dumps({"key": "a"*76 + "日本語"*5}, ensure_ascii=False))' \
    | ./target/release/rtk json -
Before: panicked at `src/cmds/system/json_cmd.rs:109`: `end byte index 77 is not a char boundary; it is inside '日'`
After: pretty-printed compact JSON with the value truncated to "aaaa…aaaa...".

### CODE REVIEW

**Comment by pszymkowiak** [comment]
> [w] **wshm** · Automated triage by AI

## 📊 Automated PR Analysis

| | |
|---|---|
| 🐛 **Type** | `bug-fix` |
| 🟢 **Risk** | `low` |

### Summary

Fixes a UTF-8 panic in the JSON compact display when truncating long string values containing multi-byte characters. Uses `str::floor_char_boundary` to round the truncation index down to a valid char boundary instead of slicing at a fixed byte offset. Includes a regression test for the multi-byte boundary case.

### Review Checklist
- [x] Tests present
- [ ] Breaking change
- [ ] Docs updated

---
*Analyzed automatically by [wshm](https://github.com/wshm-dev/wshm)* · This is an automated analysis, not a human review.

**Comment by swithek** [comment]
hey @KuSh @aeppling I think this might be a somewhat urgent issue. I'm hitting this panic quite often with emoji/non-latin content and it's pretty much breaking my rtk workflows at the moment

File: src/cmds/system/json_cmd.rs [hunk 1]
```diff
@@ -328,4 +329,19 @@ mod tests {
...
+    fn test_compact_long_multibyte_string_does_not_panic() {
+        let payload = "a".repeat(76) + &"日本語".repeat(5);
+        let json = format!(r#"{{"key": "{}"}}"#, payload);
+
+        let output = filter_json_compact(&json, 5)
+            .expect("filter_json_compact must not error on valid JSON");
+
+        assert!(output.contains("key"));
+        assert!(
+            output.contains("..."),
```
Comment (KuSh) [inline src/cmds/system/json_cmd.rs:340]: Nitpick, at least check output lenght to be <=80 characters

If i'm not mistaken this test will truncate after the last `a` and before the first `日`, it would be better to test that it can truncate within a Japanese sentence or any other non latin1 utf8 string.

**Review by KuSh — CHANGES_REQUESTED** [review]

File: src/cmds/system/json_cmd.rs (hunk 1 above)
Comment (swithek) [reply src/cmds/system/json_cmd.rs:340]: Thanks for the review @KuSh :) I've pushed an update, let me know if there's anything else that can be improved here.

File: src/cmds/system/json_cmd.rs [hunk 2]
```diff
@@ -328,4 +329,26 @@ mod tests {
...
+            "long string should be truncated, got: {output}"
+        );
+        for line in output.lines() {
+            assert!(
+                line.chars().count() <= 80,
```
Comment (KuSh) [inline src/cmds/system/json_cmd.rs]: This works by "chance" as you're counting the total line chars whereas the code is truncating the string value at 77 byte length and terminating it with "...". With your previous test `"a".repeat(76) + &"日本語".repeat(5)` this test would fail. You should check that the "key" value IS <= 80 byte (double quote excluded)

**Review by KuSh — CHANGES_REQUESTED** [review]

File: src/cmds/system/json_cmd.rs (hunk 2 above)
Comment (swithek) [reply src/cmds/system/json_cmd.rs]: Good point 👍

**Review by KuSh — APPROVED** [review]
